### PR TITLE
User association style

### PR DESCRIPTION
**Pull request by GeorgieGirl24**
# What does this PR do?

 * Add the styleByUser Query
 * Add userQuery tests

# Where should the reviewer start?

* Drop and create Crate and Crate_Test

# How should this be manually tested?

* Pull down branch and run "npm start" in code/web and code/api to ensure that the app is running
* Ensure you have the dev tools and look into the Redux section

# Any background context you want to provide?

* This allows the userId can be used to query the Style table and return the description and image_url
* This allows the query for an id that finds the description and image_url for the Style table
What are the relevant tickets?
#47 
#66 
#62
#43 
#65

### CODE REVIEW

File: code/api/src/modules/style/model.js [hunk 1]
```diff
@@ -10,4 +10,12 @@ module.exports = function(sequelize, DataTypes) {
       type: DataTypes.STRING
     }
   })
+
+  Style.associate = function(models) {
+    Style.hasMany(models.User, {
```
Comment (Callbritton) [inline code/api/src/modules/style/model.js:15]: Dropping a note as a reminder to drop this if we don't end up using this association

File: code/api/src/modules/user/model.js [hunk 2]
```diff
@@ -22,6 +22,10 @@ module.exports = function(sequelize, DataTypes) {
 
   User.associate = function(models) {
     User.hasMany(models.Subscription)
+    User.belongsTo(models.Style, {
```
Comment (Callbritton) [inline code/api/src/modules/user/model.js:25]: We will also want to remove this association here as well if we don't end up going with the relationship route

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 3]
```diff
@@ -30,12 +38,99 @@ describe('user queries', () => {
...
-    expect(response.body.data.users.length).toBe(2)
+    expect(response.body.data.users.length).toBe(10)
     // need to close the database connection for the test
     // in setup/database.js there is a connection that needs to be closed
     done();
   })
+
+  it('gets information for a specific user', async () => {
+    const response = await request(server)
+      .post('/')
```
Comment (Callbritton) [inline code/api/src/modules/user/tests/userQuery.test.js:51]: Thank you so much for adding these tests!

**Review by SageOfCode — APPROVED** [review]

**Review by Callbritton — APPROVED** [review]
Excellent work! I left a few comments for code that will need to be removed if we end up not using a relationship for User and Styles.

File: code/api/src/seeders/4-styles.js [hunk 4]
```diff
@@ -9,6 +9,24 @@ module.exports = {
         createdAt: new Date(),
         updatedAt: new Date()
       },
+      {
```
Comment (atownse) [inline code/api/src/seeders/4-styles.js:12]: Is this the data that FE will fetch and display?

**Review by atownse — APPROVED** [review]

File: code/api/src/modules/style/tests/styleQuery.test.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+//     console.log('INFO - Server started.')
+//   })
+//
+//   it('gets information for a specific style', async (done) => {
+//     const response = await request(server)
+//       .post('/')
+//       .send({ query: '{ style(id: 1) { description image_url } }' })
+//       .expect(200)
+//     // console.log(response)
+//   })
```
Comment (bruce-gordon) [inline code/api/src/modules/style/tests/styleQuery.test.js:26]: Just a note to remove these console.log statements when we no longer need them.  :)

**Review by bruce-gordon — APPROVED** [review]

**Pull request merged**